package.json

```diff
@@ -30,6 +31,7 @@
     "node": ">=4"
   },
   "dependencies": {
+    "@types/jquery": "^2.0.39",
```

Comment:
That's exactly what I was thinking while formulating an answer. I came to no real practical aproach for this, but what can be done is to create a subfolder on the repo with an alternate package.json like "ts-jquery-once". Placing the definition there and using relative path to reference the files from the repository, This way non typescript users wont load any @types;